src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileReadinessService.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.dluhc.printapi.service
+
+import mu.KotlinLogging
+import org.apache.commons.lang3.time.StopWatch
+import org.springframework.stereotype.Service
+import uk.gov.dluhc.printapi.config.SftpProperties
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseFileMessage
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class PrintResponseFileReadinessService(
+    private val sftpService: SftpService,
+    private val sftpProperties: SftpProperties,
+    private val printMessagingService: PrintMessagingService
+) {
+
+    fun markPrintResponseFileForProcessing() {
+        val stopWatch = StopWatch.createStarted()
+        val outboundFolderPath = sftpProperties.printResponseDownloadDirectory
+        logger.info { "Finding matching print responses from directory: [$outboundFolderPath]" }
+
+        val unprocessedFiles = sftpService.identifyFilesToBeProcessed(outboundFolderPath)
+        logger.info { "Found [${unprocessedFiles.size}] unprocessed print responses" }
+
+        val filesFailedToProcess = mutableListOf<String>()
+        unprocessedFiles.forEachIndexed { index, unprocessedFileName ->
+            try {
+                sftpService.markFileForProcessing(
+                    directory = outboundFolderPath,
+                    originalFileName = unprocessedFileName
+                ).also {
+                    val messagePayload = ProcessPrintResponseFileMessage(outboundFolderPath, it)
+                    logger.info { "Submitting SQS message for file: [${index + 1} of ${unprocessedFiles.size}] with payload: $messagePayload" }
+                    printMessagingService.submitPrintResponseFileForProcessing(messagePayload)
+                }
+            } catch (e: RuntimeException) {
+                filesFailedToProcess.add(unprocessedFileName)
```

Comment:
What do we do with the `filesFailedToProcess` collection? Other than add a filename to it?
Do we have a way of (or a plan to) re-process these in some way?

Reply:
Doh! I've just realized the reprocessing will happen next time the cron runs, as it will find the failed file(s) again (by virtue of the fact it did not rename this via this processing) 👍  I like it 👍 

(though does make me question more the relevance of adding the filenames to a list, other then being able to log them at the end 🤔 )